lib/sandbox/pmapi-setup-runner.js

```diff
@@ -104,12 +104,16 @@ module.exports = function (pm, onAssertionComplete) {
         }
         // if the assertion function does not expect a callback, we synchronously execute the same
         else {
-            try { assert(); }
+            try { 
+                Promise.resolve(assert())
```

Comment:
The vast majority of tests aren't going to be async functions, and we'll likely hit a small performance penalty on running all test functions through Promise resolution.  Wondering if we could do something like...

```js
try {
  const maybePromise = assert();
  if (maybePromise instanceof Promise) {
    Promise.resolve(maybePromise)
      .catch(e => markAssertionAsFailure(assertionData, e));
  }
} catch (e) {
  markAssertionAsFailure(assertionData, e);
}

onAssertionComplete(assertionData);
```

Alternatively, we could also check if the function's constructor name is `AsyncFunction` and split logic based on that without having to execute the function first.